src/main/java/com/dsrm/dsrmbackend/services/impl/UserServiceImpl.java

```diff
@@ -35,4 +36,20 @@ public Page<User> getUsers(Pageable pageable){
         return userRepo.findAll(pageable);
     }
 
+    @Override
+    public Optional<User> updateUser(UserRolesOnlyDTO userRolesOnlyDTO, Long id) {
+        final Optional<User> user = userRepo.findById(id);
+        user.ifPresent(userEntity -> updateUserRoles(userRolesOnlyDTO, userEntity));
+        return user;
+    }
+
+    @Override
```

Comment:
I would add @Transactional here because we are fetching roles and saving user.

Reply:
I added @Transactional annotation